src/App.tsx

```diff
@@ -1,61 +1,142 @@
+import { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoWithUser } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+export const initialTodos = todosFromServer.map(todo => ({
+  ...todo,
+  user: usersFromServer.find(({ id }) => id === todo.userId) || null,
+}));
 
 export const App = () => {
+  const [todoTitle, setTodoTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [userSelectedId, setUserSelectedId] = useState(0);
+  const [hasUserSelectedId, setHasUserSelectedId] = useState(false);
+  const [preparedTodos, setPreparedTodos] = useState(initialTodos);
+
+  function getUserById(userId: number) {
+    return usersFromServer.find(({ id }) => id === userId) || null;
+  }
+
+  function onSubmit(todo: TodoWithUser) {
+    setPreparedTodos(currentTodo => [...currentTodo, todo]);
+  }
+
+  function resetForm() {
+    setTodoTitle('');
+    setUserSelectedId(0);
+  }
+
+  function handleInputChange(event: React.ChangeEvent<HTMLInputElement>) {
+    setTodoTitle(event.target.value);
+    setHasTitleError(false);
+  }
+
+  function handleUserSelect(event: React.ChangeEvent<HTMLSelectElement>) {
+    setUserSelectedId(+event.target.value);
+    setHasUserSelectedId(false);
+  }
+
+  const handleOnSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!todoTitle || !userSelectedId) {
+      setHasTitleError(Boolean(!todoTitle));
+      setHasUserSelectedId(Boolean(!userSelectedId));
```

Comment:
```suggestion
      setHasTitleError(!todoTitle);
      setHasUserSelectedId(!userSelectedId);
```
Operator `not` - `!` already converted to boolean.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_NOT